### PR TITLE
GitHub action get-commit-id

### PR DESCRIPTION
**Pull request by pbchekin**
Reusable GitHub action to get the latest commit ID for the specified branch and repository without cloning the repository. We use this code in several workflows and the old implementation does not report an error in the case of networking issue. The action adds retries and error detection.

### CODE REVIEW

**Review by gshimansky — APPROVED** [review]
I am just curious, are these retries needed often?

**Comment by pbchekin** [comment]
> I am just curious, are these retries needed often?

It's not often, may be once a week. The biggest problem was that the error wasn't detected:

```
COMMIT_ID=$(git ls-remote ...)
```

In the case of network error (for example, TLS negotiation failed, unknown reason) `COMMIT_ID` was empty string and it was really hard to understand what happened.

File: .github/actions/get-commit-id/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    description: Commit id
+    value: ${{ steps.commit_id.outputs.commit_id }}
+runs:
+  using: "composite"
+  steps:
+    - name: Get commit id for ${{ inputs.repository }} ${{ inputs.branch }}
+      id: commit_id
+      shell: bash
+      run: |
+        for ((retry=0; retry<${{ inputs.retries }}; retry++)); do
```
Comment (leshikus) [inline .github/actions/get-commit-id/action.yml:28]: This is generally a good idea to add some sleep before retry happens

Comment (pbchekin) [reply .github/actions/get-commit-id/action.yml:28]: There is a sleep in line 38.

**Pull request merged**